From: Tomasz Maciej Nowak <tmn505@gmail.com>
Subject: mtd: spi-nor: winbond: w25q64: set locking feature

Wyse WLoader proprietary bootloader sets locking bit, thus only reading
from the chip is possible. Set locking feature for w25q64 in driver
to enable clearing locking bit.

--- a/drivers/mtd/spi-nor/winbond.c
+++ b/drivers/mtd/spi-nor/winbond.c
@@ -100,6 +100,7 @@ static const struct flash_info winbond_n
 	{ "w25x64", INFO(0xef3017, 0, 64 * 1024, 128)
 		NO_SFDP_FLAGS(SECT_4K) },
 	{ "w25q64", INFO(0xef4017, 0, 64 * 1024, 128)
+		FLAGS(SPI_NOR_HAS_LOCK | SPI_NOR_HAS_TB)
 		NO_SFDP_FLAGS(SECT_4K | SPI_NOR_DUAL_READ |
 			      SPI_NOR_QUAD_READ) },
 	{ "w25q64dw", INFO(0xef6017, 0, 64 * 1024, 128)
